packages/botfuel-dialog/src/dialog-manager.js

```diff
@@ -238,46 +247,56 @@ class DialogManager extends Resolver {
         data: {},
       });
     } else {
-      const action = await this.resolve(dialog.name).execute(adapter, userMessage, dialog.data);
+      const { action, botMessages: newBotMessages } = await this.resolve(dialog.name).execute(
+        userMessage,
+        dialog.data,
+      );
+      botMessages = botMessages.concat(newBotMessages);
+
       logger.debug('execute: action', action);
       if (action.name !== Dialog.ACTION_WAIT) {
```

Comment:
change condition such that return is done first

Reply:
done